packages/hash/api/src/graphql/resolvers/knowledge/page/page.ts

```diff
@@ -70,7 +70,14 @@ export const persistedPages: ResolverFn<
   QueryPersistedPagesArgs
 > = async (_, { ownedById }, { dataSources: { graphApi }, userModel }) => {
   const accountModel = ownedById
-    ? await UserModel.getUserById(graphApi, { entityId: ownedById })
+    ? await UserModel.getUserById(graphApi, { entityId: ownedById }).catch(
+        (error: Error) => {
+          if (error.message.includes("not a workspace user")) {
```

Comment:
This worries me. Can we at least have this string defined as a constant on the UserModel or something so we can import it here?

Ideally our error would actually be strongly typed with a type guard... but I'm guessing you'll argue that's a bigger task to accomplish right now haha

Reply:
I agree with you. Rather than depend on the string, I think it is actually easier to introduce a type-guard as you suggest. I've introduced a `TypeMismatchError` (let me know if you have a better idea of what to call this), and added a safer error check in https://github.com/hashintel/hash/pull/1324/commits/cf14848f2d7a19abbb9478bc1c374a2b13184516

Reply:
This looks much better, and I'm glad to see it in more places. I've added a comment inline for a further idea so I'm going to resolve this